src/profile_service/activity.details.js

```diff
@@ -1,26 +1,53 @@
 'use strict';
 
-const Age = require('../yoti_common/age').Age;
-const Profile = require('./profile').Profile;
+const { Age } = require('../yoti_common/age');
+const { Profile } = require('./profile');
+const { ApplicationProfile } = require('./application.profile');
 
-module.exports.ActivityDetails = class ActivityDetails {
-  constructor(parsedResponse, decryptedProfile) {
+/**
+ * Processes profile array data into object.
+ *
+ * @param {array} profile
+ * @returns {object}
+ */
+function parseProfile(profile) {
+  if (!profile) {
+    return {};
+  }
+  return profile.reduce((acc, current) => {
+    const propName = Object.getOwnPropertyNames(current)[0];
+    acc[propName] = current[propName];
+    return acc;
+  }, {});
+}
+
+/**
+ * Details of an activity between a user and the application.
```

Comment:
I was thinking this could be "an application" instead of "the application", but I see that it is like this in Java, so probably best to leave as is